assets/js/vueListComponent.js

```diff
@@ -0,0 +1,82 @@
+Vue.component('list', {
+    props: ["view", "filters"],
+    data() {
+        return {
+            data: [],
+            list: [],
+        }
+    },
+    methods: {
+        shuffle() {
+            // http://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle#The_modern_algorithm
+            for (var i = 0; i < this.data.length; i++) {
+                var j = i + Math.floor(Math.random() * (this.data.length - i));
+
+                var temp = this.data[j];
+                this.data[j] = this.data[i];
+                this.data[i] = temp;
+            }
+            this.list = this.data;
+        },
+        filterList() {
+            if (this.filters.type != "All") {
+                this.list = this.data.filter((itm) => {
+                    if (itm.type == null) {
+                        return false
+                    }
+                    return itm.type.toUpperCase() == this.filters.type.toUpperCase()
+                })
+            } else {
+                this.list = this.data
+            }
+
+            if (this.filters.cuisine != "All") {
+                this.list = this.list.filter((itm) => {
+                    if (itm.cuisine == null) {
+                        return false
+                    }
+                    return itm.cuisine.toUpperCase() == this.filters.cuisine.toUpperCase()
+                })
+            }
+
+            if (this.filters.region != "All") {
+                this.list = this.list.filter((itm) => {
+                    if (itm.region == null) {
+                        return false
+                    }
+                    return itm.region.toUpperCase() == this.filters.region.toUpperCase()
+                })
+            }
+
+            if (this.filters.price != "All") {
+                this.list = this.list.filter((itm) => {
+                    if (itm.price == null) {
+                        return false
+                    }
+                    return itm.price.toUpperCase() == this.filters.price.toUpperCase()
```

Comment:
It looks like you have a lot of common logic around filters. Why not create a function to reduce the repeated work?